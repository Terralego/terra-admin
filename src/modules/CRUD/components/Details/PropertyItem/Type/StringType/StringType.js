import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import NoValue from '../../NoValue';

const isHTML = value => {
  const trimmedValue = `${value}`.trim();
  if (!trimmedValue) {
    return false;
  }
  const div = document.createElement('div');
  div.innerHTML = trimmedValue;
  return div.firstChild.nodeType === Node.ELEMENT_NODE;
};

const StringType = ({ display_value: displayValue }) => {
  if (displayValue === null) {
    return <NoValue />;
  }
  return (
    <div
      className={classnames({
        details__RTE: isHTML(displayValue),
        details__text: !isHTML(displayValue),
      })}
    // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: displayValue }}
    />
  );
};

StringType.propTypes = {
  display_value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

StringType.defaultProps = {
  display_value: undefined,
};

export default StringType;
