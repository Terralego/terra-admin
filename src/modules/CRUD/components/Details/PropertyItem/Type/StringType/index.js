import StringType from './StringType';

export default StringType;
