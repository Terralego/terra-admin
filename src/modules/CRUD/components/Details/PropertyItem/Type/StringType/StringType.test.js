import React from 'react';
import renderer from 'react-test-renderer';
import StringType from './StringType';

jest.mock('../../NoValue', () => () => <div>No value</div>);

describe('should render correctly', () => {
  it('number', () => {
    const tree = renderer.create((
      <StringType display_value={42} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('text', () => {
    const tree = renderer.create((
      <StringType display_value="Foo" />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('HTML', () => {
    const tree = renderer.create((
      <StringType display_value="<p>Bar</p>" />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('nothing', () => {
    const tree = renderer.create((
      <StringType display_value={null} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
