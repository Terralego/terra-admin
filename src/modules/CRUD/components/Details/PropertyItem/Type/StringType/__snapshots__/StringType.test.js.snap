// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly HTML 1`] = `
<div
  className="details__RTE"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Bar</p>",
    }
  }
/>
`;

exports[`should render correctly nothing 1`] = `
<div>
  No value
</div>
`;

exports[`should render correctly number 1`] = `
<div
  className="details__text"
  dangerouslySetInnerHTML={
    Object {
      "__html": 42,
    }
  }
/>
`;

exports[`should render correctly text 1`] = `
<div
  className="details__text"
  dangerouslySetInnerHTML={
    Object {
      "__html": "Foo",
    }
  }
/>
`;
